filament/src/components/RenderableManager.h

```diff
@@ -106,13 +106,14 @@ class FRenderableManager : public RenderableManager {
     inline void setSkinning(Instance instance, bool enable) noexcept;
     inline void setMorphing(Instance instance, bool enable) noexcept;
     inline void setPrimitives(Instance instance, utils::Slice<FRenderPrimitive> const& primitives) noexcept;
-    inline void setBones(Instance instance, Bone const* transforms, size_t boneCount, size_t offset = 0) noexcept;
-    inline void setBones(Instance instance, math::mat4f const* transforms, size_t boneCount, size_t offset = 0) noexcept;
+    inline void setBones(Instance instance, Bone const* transforms, size_t boneCount, size_t offset = 0);
+    inline void setBones(Instance instance, math::mat4f const* transforms, size_t boneCount, size_t offset = 0);
     inline void setSkinningBuffer(Instance instance, FSkinningBuffer* skinningBuffer,
-            size_t count, size_t offset) noexcept;
-    void setMorphWeights(Instance instance, float const* weights, size_t count) noexcept;
+            size_t count, size_t offset);
+    void setMorphWeights(Instance instance, float const* weights, size_t count, size_t offset);
     void setMorphTargetBufferAt(Instance instance,
             size_t primitiveIndex, FMorphTargetBuffer* morphTargetBuffer) noexcept;
```

Comment:
Reading this code again, I think `setMorphTargertBufferAt` should work like `setGeometryAt`. 

So I think it should have `uint8_t level` parameter, which for now will always be 0, because we dont support LODs yet. But the API should have it.

This API should also have a `size_t offset, size_t count`, which define a range *inside* the `MorphTargetBuffer`. Then you need to check that:

1. this range fits in the `MorphTargetBuffer`
2. the count matches the renderable's morph target count
3. make the appropriate changes in the shader so that the offset is taken into account

This will make the API more similar to `setGeometryAt` and it will allow us to use a larger `MorphTargetBuffer` for several primitives (or in the future for lods).

I think this change should actually be done first and independently from this PR. This is my bad for not catching this before.

Please let me know what you think. Thank you!

Reply:
@pixelflinger Thanks. I agree `setMorphTargetBufferAt` shoud be work like `setGeometryAt` but I have some different idea.

I think `setMorphTargetBufferAt` should be changed to `setMorphTargetsAt` like `setGeometryAt`. Now we can know how many morph targets are needed when `Renderable` is created. Thus instead of creating and setting `MorphTargetBuffer` outside of `RenderPrimitive` , we can create `MorphTargetBuffer` in `RenderPrimitive` and update morph target data like `setGeometryAt`.

Your idea is great that we can share large `MorphTargetBuffer` with many `RenderPrimitive`s but I'm not sure sharing `MorphTargetBuffer` will be very rare and I guess it will be not happend. Because internally `MorphTargetBuffer`'s resource type is `Texture` not `Buffer`. If it is `Buffer` your idea will be very useful.

Please let me know your idea. Thank you!

Reply:
@pixelflinger Sorry ignore my previous comment. Yes your idea is correct. The vertex count exactly same at geometry and morphing. Thanks.